openedx/features/philu_courseware/helpers.py

```diff
@@ -116,3 +124,38 @@ def is_post_assessment(section):
 
 def get_section_format(section):
     return section.format.lower() if section and section.format else ''
+
+
+def validate_problem_id(problem_id):
+    """
+    validate if problem_id is valid UsageKeyField or not
+    """
+    if not problem_id:
+        raise serializers.ValidationError({'problem_id': _('Problem id is required')})
+    try:
+        return UsageKey.from_string(problem_id)
+    except InvalidKeyError:
+        raise serializers.ValidationError({'problem_id': constants.INVALID_PROBLEM_ID_MSG})
```

Comment:
Should we translate API error messages, are they user facing? If so do we need to translate `constants.INVALID_PROBLEM_ID_MSG` as well?

Reply:
Yes I think we should translate every string, either it is user facing for now or not. If we are sending in response it means it can be used either now or may be later. I have translated Invalid problem id message and also removed its constant as it is being used only at one place and secondly it is self descriptive string.